### PR TITLE
Make PodDisruptionBudget master label selector optional

### PR DESCRIPTION
**Pull request by b1zzu**
Fix: https://github.com/zalando/postgres-operator/issues/1663

Introduce the `pdb_master_label_selector` config that by default is enabled but if disabled will not add the `spilo-role: master` match label to the cluster PodDisruptionBudget.

### CODE REVIEW

**Comment by b1zzu** [comment]
Hi,
I'm open to feedback for this PR, also I didn't know if they helm charts and manifests are autogenerated or if I have to take care of it

**Comment by elliottneilclark** [comment]
This is awesome and would help us to use PDBs with Postgres.
(bumping to the top of the review pile?)

**Comment by yoshihikoueno** [comment]
@FxKu @Jan-M 
Hi, I found this MR very useful and I'm sure many users out there also find the same way.  
Could you have time to take a look at it?

File: pkg/util/config/config.go [hunk 1]
```diff
@@ -219,6 +219,7 @@ type Config struct {
 	ReplicaDNSNameFormat                     StringTemplate    `name:"replica_dns_name_format" default:"{cluster}-repl.{namespace}.{hostedzone}"`
 	ReplicaLegacyDNSNameFormat               StringTemplate    `name:"replica_legacy_dns_name_format" default:"{cluster}-repl.{team}.{hostedzone}"`
 	PDBNameFormat                            StringTemplate    `name:"pdb_name_format" default:"postgres-{cluster}-pdb"`
+	PDBMasterLabelSelector                   *bool             `name:"pdb_master_label_selector" default:"true"`
```
Comment (Jan-M) [inline pkg/util/config/config.go:222]: In line with the others why not call it enable?

**Comment by Jan-M** [comment]
@FxKu sounds like something useful to many people, tiny comment and then looks like we can tackle it.

I think it is important though for readers to understand that with this, you hand of to your infrastructure the responsibility to migrate masters only in healthy situations.

File: pkg/apis/acid.zalan.do/v1/crds.go [hunk 2]
```diff
@@ -1420,7 +1423,7 @@ var OperatorConfigCRDResourceValidation = apiextv1.CustomResourceValidation{
 										},
 									},
 								},
-							},
+              },
```
Comment (FxKu) [inline pkg/apis/acid.zalan.do/v1/crds.go:1426]:
```suggestion
},
```

Comment (FxKu) [inline pkg/apis/acid.zalan.do/v1/crds.go:1426]:
```suggestion
							},
```

**Comment by FxKu** [comment]
:+1:

**Comment by idanovinda** [comment]
👍

**Pull request merged**